meilisearch-http/src/index_controller/uuid_resolver/mod.rs

```diff
@@ -33,20 +33,37 @@ pub trait UuidResolverHandle {
     async fn list(&self) -> anyhow::Result<Vec<(String, Uuid)>>;
     async fn snapshot(&self, path: PathBuf) -> Result<HashSet<Uuid>>;
     async fn get_size(&self) -> Result<u64>;
+    async fn dump(&self, path: PathBuf) -> Result<HashSet<Uuid>>;
 }
 
 #[derive(Debug, Error)]
-pub enum UuidError {
+pub enum UuidResolverError {
     #[error("Name already exist.")]
     NameAlreadyExist,
     #[error("Index \"{0}\" doesn't exist.")]
     UnexistingIndex(String),
-    #[error("Error performing task: {0}")]
-    TokioTask(#[from] tokio::task::JoinError),
-    #[error("Database error: {0}")]
-    Heed(#[from] heed::Error),
-    #[error("Uuid error: {0}")]
-    Uuid(#[from] uuid::Error),
     #[error("Badly formatted index uid: {0}")]
     BadlyFormatted(String),
+    #[error("Internal error resolving index uid: {0}")]
+    Internal(String),
 }
+
+macro_rules! internal_error {
+    ($($other:path), *) => {
+        $(
+            impl From<$other> for UuidResolverError {
+                fn from(other: $other) -> Self {
+                    Self::Internal(other.to_string())
+                }
+            }
+        )*
+    }
+}
+
```

Comment:
Why do we copy-paste this code in every actor instead of defining once for everyone and call it everywhere?

Reply:
because it's a bit of a workaround for now, it gets things done until I work on error handling